leftwm-core/src/display_servers/xlib_display_server/event_translate.rs

```diff
@@ -70,12 +72,26 @@ fn from_destroy_notify(x_event: XEvent) -> Option<DisplayEvent> {
     let event = xlib::XDestroyWindowEvent::from(x_event.1);
     if xw.managed_windows.contains(&event.window) {
         let h = event.window.into();
-        xw.teardown_managed_window(&h);
+        xw.teardown_managed_window(&h, true);
         return Some(DisplayEvent::WindowDestroy(h));
     }
     None
 }
 
+fn from_focus_in(x_event: XEvent) -> Option<DisplayEvent> {
+    let xw = x_event.0;
+    let event = xlib::XFocusChangeEvent::from(x_event.1);
+    // Check that if a window is taking focus, that it should be.
+    if xw.focused_window != event.window {
+        let never_focus = match xw.get_wmhints(xw.focused_window) {
+            Some(hint) => hint.flags & xlib::InputHint != 0 && hint.input == 0,
+            None => false
```

Comment:
Sorry, my suggestion tripped up fmt :smile: 

```suggestion
            None => false,
```